checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInference.java

```diff
@@ -367,41 +438,65 @@ private void analyzeOwnershipTransferAtAssignment(
       if (!TreeUtils.isConstructor(methodTree)) {
         disposedFields.remove((VariableElement) lhsElement);
       }
-      addOwningToParamsIfDisposedAtAssignment(obligations, rhsObligation, rhs);
```

Comment:
Why were the two calls to this method (here and at line 372 below) removed? These changes don't seem necessary (maybe they are unintentional? the only parts of these changes that seem required are the new lines 447 to 453, based on the Javadoc changes)

Reply:
This change was intentional. I changed `addOwningToParamsIfDisposedAtAssignment` method to `getIndexOfParam` to reduce the redundant computations in this method and also in the `analyzeReturnNode`. The new method returns the index of formal parameter that is alias with the given obligation. In this method, the given obligation is the rhs obligation, and in the  `analyzeReturnNode` is the obligation of the return node.